code/modules/projectiles/guns/projectile/constructable/gunsmithing.dm

```diff
@@ -743,6 +743,80 @@
 			update_wheelchair_assembly()
 //CANNON END///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+//WIND-UP BEGIN////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/item/weapon/windup_assembly
+	name = "wind-up assembly"
+	desc = "A spring in a shoe and all the associated possibilities."
+	icon = 'icons/obj/weaponsmithing.dmi'
+	icon_state = "windup_assembly_0"
+	var/windupstate = 0
+	var/list/windup_adhesive = list(
+		/obj/item/gum,
+		/obj/item/taperoll/police,
+		/obj/item/taperoll/engineering,
+		/obj/item/taperoll/atmos,
+		/obj/item/taperoll/syndie,
+		/obj/item/device/label_roll,
+		/obj/item/weapon/glue,
+		/obj/item/weapon/bonegel,
+		/obj/item/stack/package_wrap,
+		/obj/item/stack/medical/bruise_pack/bandaid,
+		/obj/item/weapon/reagent_containers/food/snacks/caramelburger
+		)
+
+/obj/item/weapon/stock_parts/spring/attackby(var/obj/item/I, mob/user)
+	if(istype(I, /obj/item/clothing/shoes))
+		var/obj/item/clothing/shoes/G = I
+		playsound(src,'sound/effects/spring.ogg', 50,1)
+		var/obj/item/weapon/windup_assembly/W = new /obj/item/weapon/windup_assembly
```

Comment:
make this `= new(get_turf(src))`

that way it acts as a fallback in case the user's hands are somehow full (shouldn't, but it's good to be safe against nullspacing)